packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -264,6 +283,25 @@ export function getChildFormatContext(
     return createFormatContext(HTML_MODE, null);
   }
   if (parentContext.insertionMode === ROOT_HTML_MODE) {
+    switch (type) {
+      case 'html': {
+        return parentContext;
```

Comment:
This kind of changes the meaning of ROOT_HTML_MODE too since it's now also applicable one level down. E.g. the doctype being emitted isn't quite the same now as it'll be emitted again for another `<html>` (which is invalid regardless).